wgpu-core/src/command/bundle.rs

```diff
@@ -1036,14 +1040,20 @@ pub mod bundle_ffi {
         stages: wgt::ShaderStage,
         offset: u32,
         size_bytes: u32,
-        data: *const u32,
+        data: *const u8,
     ) {
         span!(_guard, DEBUG, "RenderBundle::set_push_constants");
-        let data_slice = slice::from_raw_parts(data, (size_bytes / 4) as usize);
+        let data_slice = slice::from_raw_parts(data, size_bytes as usize);
         let value_offset = pass.base.push_constant_data.len().try_into().expect(
             "Ran out of push constant space. Don't set 4gb of push constants per RenderBundle.",
         );
+
+        let aligned_len = ((data_slice.len() + 3) / 4) * 4;
```

Comment:
not sure if the codegen will be different, but it may be better to write as `(((data_slice.len() - 1) | 3) + 1)`